#!/usr/bin/env python
PACKAGE = "auv_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# PD Gains for X (horizontal) axis
gen.add("kp_x", double_t, 0, "Proportional gain for X axis", 0.01, 0.0, 1.0)
gen.add("kd_x", double_t, 0, "Derivative gain for X axis", 0.01, 0.0, 1.0)

# PD Gains for Y (vertical) axis
gen.add("kp_y", double_t, 0, "Proportional gain for Y axis", 0.01, 0.0, 1.0)
gen.add("kd_y", double_t, 0, "Derivative gain for Y axis", 0.01, 0.0, 1.0)

# PD Gains for angular Z (yaw) axis
gen.add("kp_yaw", double_t, 0, "Proportional gain for angular Z", 0.1, 0.0, 2.0)
gen.add("kd_yaw", double_t, 0, "Derivative gain for angular Z", 0.05, 0.0, 1.0)

# Max wrench limits
gen.add("max_force_x", double_t, 0, "Max force in X direction (N)", 50.0, 0.0, 200.0)
gen.add("max_force_y", double_t, 0, "Max force in Y direction (N)", 50.0, 0.0, 200.0)
gen.add("max_torque_z", double_t, 0, "Max torque in Z direction (Nm)", 20.0, 0.0, 100.0)

exit(gen.generate(PACKAGE, "auv_control", "BottomVisualServoing"))
